pyccl/cosmology.py

```diff
@@ -694,7 +694,7 @@ def _get_halo_model_nonlin_power(self):
         prf = hal.HaloProfileNFW(concentration=c)
         hmc = hal.HMCalculator(mass_function=hmf, halo_bias=hbf,
                                mass_def=mdef)
-        return hal.halomod_Pk2D(self, hmc, prf)
+        return hal.halomod_Pk2D(self, hmc, prf, normprof1=True)
```

Comment:
without this, the halo model power spectrum as a Cosmology power spectrum option (which will be deprecated in v3) gave the wrong answer.